esphome/components/esp32_improv/esp32_improv_component.h

```diff
@@ -54,6 +64,8 @@ class ESP32ImprovComponent : public Component, public BLEServiceComponent {
   void set_wifi_timeout(uint32_t wifi_timeout) { this->wifi_timeout_ = wifi_timeout; }
   uint32_t get_wifi_timeout() const { return this->wifi_timeout_; }
 
+  improv::State get_improv_state() const { return this->state_; }
+
```

Comment:
```suggestion
   improv::State get_improv_error_state() const { return this->error_; }
   
```
Allow external access to the latest error state.